### PR TITLE
OGM-1073 moving constant pool size to configuration

### PR DESCRIPTION
**Pull request by schernolyas**
Moving constant pool size to parameter of configuration

### CODE REVIEW

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/Neo4jProperties.java [hunk 1]
```diff
@@ -54,6 +54,14 @@
 	 */
 	public static final String ESTABLISH_CONNECTION_TIMEOUT = "hibernate.ogm.neo4j.client.establish_connection_timeout";
 
+	/**
+	 * The size of the client connection pool used by the RestEasy HTTP client
+	 * <p>
+	 *     Default value is 10
+	 * </p>
+	 */
+	public static final String REST_CONNECTION_POOL_SIZE = "hibernate.ogm.neo4j.client.rest.connection_pool_size";
```
Comment (DavideD) [inline neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/Neo4jProperties.java]: I would prefer not to mention the term REST, let's just remove it. Same for all the other places

Comment (DavideD) [reply neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/Neo4jProperties.java]: or maybe just use http instead of rest

**Comment by schernolyas** [comment]
Hi @DavideD !

I see https://stackoverflow.com/questions/39339708/neo4j-bolt-driver-connection-pooling . I will try  configure it

**Comment by schernolyas** [comment]
Davide, class org.hibernate.ogm.datastore.neo4j.remote.bolt.impl.BoltNeo4jClient has incorrect order of import. I fix it too.

**Comment by DavideD** [comment]
> Davide, class org.hibernate.ogm.datastore.neo4j.remote.bolt.impl.BoltNeo4jClient has incorrect order of import. I fix it too.

@schernolyas in the future could you make an effort and keep this kind of changes in separate commits, please?

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/common/impl/RemoteNeo4jConfiguration.java [hunk 2]
```diff
@@ -150,4 +158,12 @@ public Long getEstablishConnectionTimeout() {
 	public boolean isAuthenticationRequired() {
 		return authenticationRequired;
 	}
+
+	/**
+	 * @see Neo4jProperties#CONNECTION_POOL_SIZE
+	 * @return the size of REST connection pool
+	 */
+	public Integer getClientRestPoolSize() {
```
Comment (DavideD) [inline neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/common/impl/RemoteNeo4jConfiguration.java]: Remove the Rest part, you can call it ClientPoolSize().

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jClient.java [hunk 3]
```diff
@@ -35,13 +38,15 @@ public BoltNeo4jClient(RemoteNeo4jDatabaseIdentifier identifier, RemoteNeo4jConf
 
 	private Driver createNeo4jDriver(RemoteNeo4jDatabaseIdentifier identifier, RemoteNeo4jConfiguration configuration) {
 		String uri = identifier.getDatabaseUri();
+		Config.ConfigBuilder configBuilder = Config.build();
+		configBuilder.withMaxIdleSessions( configuration.getClientRestPoolSize()  );
```
Comment (DavideD) [inline neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jClient.java]: Do these two attibutes have the same meaning?

**Comment by schernolyas** [comment]
Okey @DavideD  I will try. For now, I will send commit for restore previous formatting and new commit for correct formatting

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/common/impl/RemoteNeo4jConfiguration.java (hunk 2 above)
Comment (schernolyas) [reply neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/common/impl/RemoteNeo4jConfiguration.java]: OK. thanks!

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jClient.java (hunk 3 above)
Comment (schernolyas) [reply neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jClient.java]: Hm ... What do you mean ? Code in row 42 is correct default configuration of Bolt driver.

**Comment by DavideD** [comment]
> Okey @DavideD I will try. For now, I will send commit for restore previous formatting and new commit for correct formatting

Just to clarify, the reason I'm asking to be careful when you make this changes is that they make the commit diff larger and therefore harder to review. If, in the same commit, you add new imports and move around existing ones, it becomes harder for the reviewer to understand what's new and what was already there.

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jClient.java (hunk 3 above)
Comment (DavideD) [reply neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jClient.java]: This line is setting the MaxIdleSessions (a time) using the value in ClientRestPoolSize ( a number of connections). I don't understand how the two values are connected.

Comment (schernolyas) [reply neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jClient.java]: Hm ... I am look to Javadoc
/**
         * The max number of idle sessions to keep open at once. Configure this
         * higher if you want more concurrent sessions, or lower if you want
         * to lower the pressure on the database instance.
         *
         * @param size the max number of idle sessions to keep open
         * @return this builder
         */
        public ConfigBuilder withMaxIdleSessions( int size )

It is not a time ... it is count of connections.

Comment (DavideD) [reply neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jClient.java]: Ah, OK. I probably checked the wrong property. But they still seems quite different to me. One is a pool size and the other one is a limit. I think it's better to keep two separates properties for this. Let's ignore bolt for this PR.

**Comment by schernolyas** [comment]
Hi @DavideD !
The PR is ready for review.

**Comment by DavideD** [comment]
> The PR is ready for review.

Could you remove the Bolt part? `MaxIdleSessions` is different from a connection pool.
Let's use the property only in the `HttpNeo4jClient` only.

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jClient.java (hunk 3 above)
Comment (DavideD) [reply neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/bolt/impl/BoltNeo4jClient.java]: This method is still called `getClientRestPoolSize()`, could you rename it to `getClientPoolSize()`

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/common/impl/RemoteNeo4jConfiguration.java [hunk 4]
```diff
@@ -24,6 +24,11 @@
 
 	public static final int DEFAULT_BOLT_PORT = 7687;
 
+	/**
+	 * Default Size of the client connection pool used by the RestEasy HTTP client
+	 */
+	public static final int DEFAULT_REST_CONNECTION_POOL_SIZE = 10;
```
Comment (DavideD) [inline neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/common/impl/RemoteNeo4jConfiguration.java]: Please, rename this to DEFAULT_CONNECTION_POOL_SIZE

Comment (schernolyas) [reply neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/common/impl/RemoteNeo4jConfiguration.java]: Hi @DavideD !
Ok. 5 minutes

**Comment by DavideD** [comment]
Jenkins is failing can you have a look if it's related to your PR?: http://ci.hibernate.org/job/hibernate-ogm-PR/1459/console

**Comment by DavideD** [comment]
> Jenkins is failing can you have a look if it's related to your PR?: http://ci.hibernate.org/job/hibernate-ogm-PR/1459/console

It might be unrelated

**Comment by schernolyas** [comment]
Hm .... May be ... I have fixed name of method.

**Comment by DavideD** [comment]
It seems it was a problem in CI

**Comment by schernolyas** [comment]
Hi @DavideD !
The PR is ready for review.

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-ogm/pull/911

I squashed the commits together and removed the changes to the bolt dialect.

@schernolyas Can you add the property to the documentation and send a new PR?

Thanks

**Pull request closed without merging**